server/src/main/java/org/apache/druid/segment/realtime/firehose/HttpFirehoseFactory.java

```diff
@@ -64,6 +71,20 @@ public HttpFirehoseFactory(
     final URLConnection connection = uris.get(0).toURL().openConnection();
     final String acceptRanges = connection.getHeaderField(HttpHeaders.ACCEPT_RANGES);
     this.supportContentRange = acceptRanges != null && "bytes".equalsIgnoreCase(acceptRanges);
+    this.httpAuthenticationUsername = httpAuthenticationUsername;
+    this.httpAuthenticationPasswordProvider = httpAuthenticationPasswordProvider == null ? new DefaultPasswordProvider("") : httpAuthenticationPasswordProvider;
```

Comment:
Should this have a default value? Or is it fine to set it to null if the parameter is null?

Reply:
Would you please check this comment?

Reply:
@jihoonson I defaulted to a Default PasswordProvider with empty string password in order to simplify the logic on line 213 (https://github.com/apache/incubator-druid/pull/7145/files#diff-4adc70bca196e07dfd5195379218c9e6R213). If we just leave with null I suppose we could change this logic to not throw NullPointerException. Do you think letting it would be null and changing the logic on 213 would be a better choice?

Reply:
I see. I think it's better to leave it to null because the default value would appear in the log which I think can make users confused by showing a password provider which is not set by users. What do you think?

Reply:
I agree now that I think about it. Have made the change in latest commit.